src/Language/PureScript/Constants.hs

```diff
@@ -342,6 +342,29 @@ pattern DataSymbol = ModuleName [ProperName "Data", ProperName "Symbol"]
 pattern IsSymbol :: Qualified (ProperName 'ClassName)
 pattern IsSymbol = Qualified (Just DataSymbol) (ProperName "IsSymbol")
 
+-- Data.Nat
+
+pattern DataNat :: ModuleName
+pattern DataNat = ModuleName [ProperName "Data", ProperName "Nat"]
+
+pattern IsNat :: Qualified (ProperName 'ClassName)
+pattern IsNat = Qualified (Just DataNat) (ProperName "IsNat")
+
+-- Prim.Nat
+
+pattern TypeDataNat :: ModuleName
+pattern TypeDataNat = ModuleName [ProperName "Prim", ProperName "Nat"]
+
+pattern CompareNat :: Qualified (ProperName 'ClassName)
```

Comment:
Can these move to `Prim.Nat` as well?

Reply:
I should rename this patsyn to `PrimNat`, but as far as I can tell, all the purely typelevel classes are in `Prim.Nat` now - I didn't move `IsNat`, because it has a `reifyNat` function. Wouldn't that be a problem?

Reply:
Great!

> Wouldn't that be a problem?

Yes, it would, unless we find a better way to encode things. We could always say the integer _is_ the dictionary, but then we wouldn't be able to write the class definition in such a way that people couldn't write bad instances.

Reply:
You'll also need to add these to `primClasses`, and also add docstrings for them in the corresponding `Docs` module, so that they can be linked on Pursuit.